homeassistant/components/devolo_home_control/devolo_device.py

```diff
@@ -73,3 +76,13 @@ def name(self):
     def available(self) -> bool:
         """Return the online state."""
         return self._available
+
+    def _sync(self, message):
+        """Update the binary sensor state."""
+        if message[0] == self._unique_id:
+            self._value = message[1]
+        elif message[0].startswith("hdm"):
+            self._available = self._device_instance.is_online()
```

Comment:
This block exists 4 times in the integration.  It might be nice to de-dupe this as well in a future PR.